app/assets/javascripts/discourse/app/components/add-category-tag-classes.js

```diff
@@ -15,36 +15,43 @@ export default Component.extend({
       return;
     }
     const slug = this.get("category.fullSlug");
-    const tags = this.tags;
 
     this._removeClass();
 
-    let classes = [];
+    const classes = [];
+
     if (slug) {
       classes.push("category");
       classes.push(`category-${slug}`);
     }
-    if (tags) {
-      tags.forEach((t) => classes.push(`tag-${t}`));
-    }
-    if (classes.length > 0) {
-      $("body").addClass(classes.join(" "));
-    }
+
+    this.tags?.forEach((t) => classes.push(`tag-${t}`));
+
+    document.body.classList.add(...classes);
   },
 
   @observes("category.fullSlug", "tags")
   refreshClass() {
     scheduleOnce("afterRender", this, this._updateClass);
   },
 
-  _removeClass() {
-    $("body").removeClass((_, css) =>
-      (css.match(/\b(?:category|tag)-\S+|( category )/g) || []).join(" ")
-    );
-  },
-
   willDestroyElement() {
     this._super(...arguments);
     this._removeClass();
   },
+
+  _removeClass() {
+    const invalidClasses = [];
+    const regex = /\b(?:category|tag)-\S+|( category )/g;
+
+    document.body.classList.forEach((name) => {
+      if (name.match(regex)) {
```

Comment:
Super minor thing but when you're interested only in true/false then `.test()` is the way to go:

```suggestion
      if (regex.test(name)) {
```